.github/workflows/black-formatting.yml

```diff
@@ -0,0 +1,64 @@
+# Taken from https://github.com/cclauss/autoblack 
+#-- Removed the functionality to automatically fix (can be re-added)
+#
+#
+# See https://github.com/cclauss/autoblack/blob/master/.github/workflows/autoblack_pull_request.yml
+# See also https://github.com/cclauss/autoblack/blob/master/.github/workflows/autoblack_push.yml
+
+
+name: autoblack-push
+on:
+  push:
+    branches: [ main ]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:  # https://github.com/stefanzweifel/git-auto-commit-action#checkout-the-correct-branch
+            ref: ${{ github.head_ref }}
+      - uses: actions/setup-python@v1
+      - run: pip install black
+      - run: black --check .
+#      - name: If needed, commit black changes to a new pull request
+#        if: failure()
+#        run: |
+#          black .
+#          git config --global user.name github-actions
+#          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'          
+#          # git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+#          git commit -am "fixup! Format Python code with psf/black push"
+#          git push  # --force origin HEAD:$GITHUB_REF
+
+
+name: autoblack-pull
+on:
+  pull_request:
+    branches: [ main ]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:  # https://github.com/stefanzweifel/git-auto-commit-action#checkout-the-correct-branch
+            ref: ${{ github.head_ref }}
+      - uses: actions/setup-python@v2
+      - run: pip install black
+      - run: black --check .
+#      - name: If needed, commit black changes to the pull request
+#        if: failure()
+#        run: |
+#          printenv | grep GITHUB
+#          git config --global user.name 'cclauss'
+#          git config --global user.email 'cclauss@users.noreply.github.com'          
+#          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+#          git remote -v
+#          git branch
+#          git status
+#          black .
+#          git status
+#          echo ready to commit
+#          git commit -am "fixup! Format Python code with psf/black pull_request"
+#          echo ready to push
+#          git push
+#          # git push --force origin
```

Comment:
In .github/workflows/black-formatting.yml, 
If you have already setup a Github token secret, you can uncomment these lines to allow the workflow to automatically format the code on every pull request or push
```diff
diff --git a/.github/workflows/black-formatting.yml b/.github/workflows/black-formatting.yml
new file mode 100644
index 0000000..6d1ccfe
--- /dev/null
+++ b/.github/workflows/black-formatting.yml
@@ -0,0 +1,64 @@
+# Taken from https://github.com/cclauss/autoblack
+#-- Removed the functionality to automatically fix (can be re-added)
[...]
+# See https://github.com/cclauss/autoblack/blob/master/.github/workflows/autoblack_pull_request.yml
+# See also https://github.com/cclauss/autoblack/blob/master/.github/workflows/autoblack_push.yml
[...]
+name: autoblack-push
+on:
+  push:
+    branches: [ main ]
[...]
+#      - name: If needed, commit black changes to a new pull request
+#        if: failure()
+#        run: |
+#          black .
+#          git config --global user.name github-actions
+#          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'
+#          # git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+#          git commit -am "fixup! Format Python code with psf/black push"
+#          git push  # --force origin HEAD:$GITHUB_REF
+
+name: autoblack-pull
+on:
+  pull_request:
+    branches: [ main ]
[...]
+#      - name: If needed, commit black changes to the pull request
+#        if: failure()
+#        run: |
+#          printenv | grep GITHUB
+#          git config --global user.name 'cclauss'
+#          git config --global user.email 'cclauss@users.noreply.github.com'
+#          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+#          git remote -v
+#          git branch
+#          git status
+#          black .
+#          git status
+#          echo ready to commit
+#          git commit -am "fixup! Format Python code with psf/black pull_request"
+#          echo ready to push
+#          git push
+#          # git push --force origin
```